test-integration/helpers/reading-builder.coffee

```diff
@@ -210,6 +210,10 @@ class ReadingBuilder extends TestHelper
         @test.driver.wait =>
           @el.pendingPublishButton.isPresent().then (isPresent) -> not isPresent
 
+      # Publishing can be async so wait until the assignment shows up as 'published' in the calendar
+      # Added here because folks won't remember to wait for this
+      (new Calendar.CalendarHelper(@test)).waitUntilPublishingFinishedByTitle(name)
+
   save: =>
```

Comment:
This is really good - I wrote the earlier `@el.pendingPublishButton.isPresent()` wait, but that'll fail when the publish is performed by a background job (as you discovered).